### DISCUSSION TITLE
Providing some information before FXGL initialization?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1166] [Ideas]
Hi:

Is it possible to provide some services e.g. version info. etc. before FXGL is inited?
Currently FXGL supports embedded mode, so we may use some info. or services outside the FXGL pane or before it is inited or after it is closed
so is it possible to move such services from after initialization to normal static method?
e.g.
```java
FXGL.getVersion();
```

Thanks

**Comment by chengenzhao** [comment]
and FXGL.getFileSystemService() also be nice to have
etc.

**Comment by AlmasB** [comment]
Hi. The short answer is no. FXGL functionality and services are only accessible between `init` and `exit` as per its architecture.

Changing this behavior would require a change to the architecture, which is not feasible at this stage. A workaround is presumably to start FXGL early in the application lifecycle and keep it running until it is not needed.
